enzy_htp/analysis/rmsd.py

```diff
@@ -0,0 +1,90 @@
+"""Submodule contains code for calculations the RMSD value of a StructureEnsemble instance.
+The calculation can be performed with Amber interface.
++ rmsd_calculation()
+    Calculate the RMSD value of a StructureEnsemble instance with specified mask pattern.
++ rmsd_of_region()
+    Get RMSD value from MD simulation result with a mask region (a list of residues).
++ rmsd_of_structure()
+    Get RMSD value from MD simulation result of the whole structure.
+
+Author: Zhong, Yinjie <yinjie.zhong@vanderbilt.edu>
+
+Date: 2024-11-15
+"""
+# Here put the import lib.
+from typing import List, Dict, Union
+
+# Here put EnzyHTP imports.
+from enzy_htp import interface as eh_interface
+from enzy_htp import config as eh_config
+from enzy_htp.core import file_system as fs
+from enzy_htp.preparation import remove_solvent
+from enzy_htp.structure import Residue, Structure
+from enzy_htp.structure.structure_ensemble import StructureEnsemble
+from enzy_htp.structure.structure_selection import select_stru
+
+def compose_pymol_pattern(mask_region: List[Residue] = list(), ca_only: bool = True) -> str:
+    """Compose the given list of Residue instances to PyMOL-format pattern.
+    
+    Args:
+        mask_region (List[Residue], optional): A list of Residues to calculation RMSD value. If the list is empty, then all residues are selected.
+        include_ligand (bool, optional): Indicate if ligands are included in RMSD calculation. Default True.
+        ca_only (bool, optional): Indicate if only C-alpha are included in RMSD calculation; otherwise all atoms are included. Default True.
+    
+    Returns:
+        region_pattern (str): A pymol-formatted selection string which defines the region for calculating RMSD value.
+    """
+    region_pattern = str()
+
+    if (mask_region):
+        region_pattern = f"resi {'+'.join([str(residue.idx) for residue in mask_region])}"
+    else:
+        region_pattern = "all"
+    if (ca_only):
+        region_pattern += " and name CA"
+    else:
+        region_pattern += " and (not elem H)"
+    return region_pattern
```

Comment:
We dont need to compose a pymol pattern. A list of residues will either COME from a pymol pattern or getter of a Structure which later case needs to covert into a StruSelection.
The AmberInterface should never know anything about PyMol. So the logic should be like below:
analysis/rmsd.py:
    pattern -> StruSeletion -> rmsd
    List[Residue] -> StruSelection -> rmsd

_interface/amber_interface.py:
    StruSelection -> cpptraj mask

Reply:
After all I really don't think we need API for List[Residue] and Structure in any cases. Consider removing them or reply an application scenario down here.

Reply:
I see!

Reply:
This issue is resolved.